File: .sdk-repos/orchestration-cluster-api-go/examples/recovery.go
// Recovery operations: change cluster mode and restore from backup.
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func changeClusterModeExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region ChangeClusterMode
	result, err := client.ChangeClusterMode(ctx)
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion ChangeClusterMode
	return nil
}

func restoreExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region Restore
	result, err := client.Restore(ctx, *openapi.NewRestoreRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion Restore
	return nil
}

func getRestoreStatusExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetRestoreStatus
	// Reports the in-flight restore only — 404 once it has finished.
	status, err := client.GetRestoreStatus(ctx)
	if err != nil {
		return err
	}
	fmt.Printf("restore %s: %s\n", status.GetChangeId(), status.GetStatus())
	for _, broker := range status.GetBrokers() {
		fmt.Printf("%v\n", broker)
	}
	// endregion GetRestoreStatus
	return nil
}

func changeClusterModeAsClusterAdminExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region ChangeClusterModeAsClusterAdmin
	// Changes the cluster mode as a cluster-level admin (cross-tenant authority).
	result, err := client.ChangeClusterModeAsClusterAdmin(ctx, func(r openapi.ApiChangeClusterModeAsClusterAdminRequest) openapi.ApiChangeClusterModeAsClusterAdminRequest {
		return r.Mode(openapi.MODE_RECOVERING)
	})
	if err != nil {
		return err
	}
	fmt.Printf("change %s: %d planned operation group(s)\n", result.GetChangeId(), len(result.GetPlannedChanges()))
	// endregion ChangeClusterModeAsClusterAdmin
	return nil
}

func restoreAsClusterAdminExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region RestoreAsClusterAdmin
	// Triggers a cluster-level restore (cluster-admin authority), restoring from the given backup IDs.
	// backupIds are one per partition, so the placeholder slice below must be extended to
	// match the actual partition count of the target cluster (shown here for a 2-partition cluster).
	restoreRequest := openapi.NewClusterRestoreRequestWithDefaults()
	restoreRequest.SetBackupIds([]int64{1, 2})
	result, err := client.RestoreAsClusterAdmin(ctx, *restoreRequest)
	if err != nil {
		return err
	}
	fmt.Printf("restore change id: %s\n", result.GetChangeId())
	// endregion RestoreAsClusterAdmin
	return nil
}

File: .sdk-repos/orchestration-cluster-api-php/examples/recovery.php
<?php

/**
 * Compilable usage examples for cluster recovery operations.
 *
 * Recovery actions affect the cluster globally and should be run only by an authorized
 * operator with a verified recovery plan.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region ChangeClusterMode
/**
 * Change cluster mode.
 */
function change_cluster_mode(CamundaClient $client, \Camunda\Orchestration\Api\Model\Mode $mode, ?bool $dryRun = false): void
{
    $client->changeClusterMode($mode, $dryRun);
}
// endregion ChangeClusterMode

// region Restore
/**
 * Restore from a backup.
 */
function restore(CamundaClient $client, \Camunda\Orchestration\Api\Model\RestoreRequest $restoreRequest, ?bool $dryRun = false): void
{
    $client->restore($restoreRequest, $dryRun);
}
// endregion Restore

// region GetRestoreStatus
/**
 * Get the status of the in-progress restore.
 */
function get_restore_status(CamundaClient $client): void
{
    $client->getRestoreStatus();
}
// endregion GetRestoreStatus

// region ChangeClusterModeAsClusterAdmin
/**
 * Change cluster mode as cluster admin.
 * @param array<string, mixed> $variables
 */
function change_cluster_mode_as_cluster_admin(CamundaClient $client, \Camunda\Orchestration\Api\Model\Mode $mode, ?string $physicalTenantId = null, ?bool $dryRun = false, ?int $hostIndex = null, array $variables = []): void
{
    $client->changeClusterModeAsClusterAdmin($mode, $physicalTenantId, $dryRun, $hostIndex, $variables);
}
// endregion ChangeClusterModeAsClusterAdmin

// region RestoreAsClusterAdmin
/**
 * Restore as cluster admin.
 * @param array<string, mixed> $variables
 */
function restore_as_cluster_admin(CamundaClient $client, \Camunda\Orchestration\Api\Model\ClusterRestoreRequest $clusterRestoreRequest, ?string $physicalTenantId = null, ?bool $dryRun = false, ?int $hostIndex = null, array $variables = []): void
{
    $client->restoreAsClusterAdmin($clusterRestoreRequest, $physicalTenantId, $dryRun, $hostIndex, $variables);
}
// endregion RestoreAsClusterAdmin
